src/dcnum/write/queue_collector_base.py

```diff
@@ -104,18 +106,19 @@ def require_feature(self, feat, sample_data):
         return self.events[feat]
 
 
-class QueueCollectorThread(threading.Thread):
+class QueueCollectorBase:
     def __init__(self,
                  event_queue: mp.Queue,
                  writer_dq: deque,
```

Comment:
If `QueueCollectorBase` is the only class that needs to access `writer_dq`, then the `writer_dq` argument should be removed and it should be initialized in `run`. The point of adding `writer_queue_length` was that we cannot share `writer_dq` between processes.

Remove the `writer_dq` kwarg and initialize it in `run` instead. Don't forget to remove it from anything that initializes `QueueCollectorBase`.